integration/setup/tenant_cluster.go

```diff
@@ -295,3 +338,503 @@ func ensureCertConfigsInstalled(ctx context.Context, id string, config Config) e
+				{
+					Name:   aws.String("tag:aws:cloudformation:logical-id"),
+					Values: []*string{aws.String("WorkerSecurityGroup")},
+				},
+			},
+		}
+
+		o, err := config.AWSClient.EC2.DescribeSecurityGroups(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		if len(o.SecurityGroups) != 1 {
+			return microerror.Maskf(executionFailedError, "expected one security group, got %d", len(o.SecurityGroups))
+		}
+
+		workerSecurityGroupID = *o.SecurityGroups[0].GroupId
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("waited for worker security group %#q", workerSecurityGroupID))
+	}
+
+	var bastionSecurityGroupID string
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "creating bastion security group")
+
+		i := &ec2.CreateSecurityGroupInput{
+			Description: aws.String("Allow SSH access from everywhere to port 22."),
+			GroupName:   aws.String(clusterID + "-bastion"),
+			VpcId:       aws.String(vpcID),
+		}
+
+		o, err := config.AWSClient.EC2.CreateSecurityGroup(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		bastionSecurityGroupID = *o.GroupId
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("created bastion security group %#q", bastionSecurityGroupID))
+	}
+
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "tagging bastion security group")
+
+		i := &ec2.CreateTagsInput{
+			Resources: []*string{
+				aws.String(bastionSecurityGroupID),
+			},
+			Tags: []*ec2.Tag{
+				{
+					Key:   aws.String("Name"),
+					Value: aws.String(clusterID + "-bastion"),
+				},
+				{
+					Key:   aws.String("giantswarm.io/cluster"),
+					Value: aws.String(clusterID),
+				},
+			},
+		}
+
+		_, err = config.AWSClient.EC2.CreateTags(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "tagged bastion security group")
+	}
+
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "updating bastion security group to allow ssh access")
+
+		i := &ec2.AuthorizeSecurityGroupIngressInput{
+			GroupId: aws.String(bastionSecurityGroupID),
+			IpPermissions: []*ec2.IpPermission{
+				{
+					FromPort:   aws.Int64(22),
+					IpProtocol: aws.String("tcp"),
+					IpRanges: []*ec2.IpRange{
+						{
+							CidrIp:      aws.String("0.0.0.0/0"),
+							Description: aws.String("Allow SSH access from everywhere to port 22."),
+						},
+					},
+					ToPort: aws.Int64(22),
+				},
+			},
+		}
+
+		_, err = config.AWSClient.EC2.AuthorizeSecurityGroupIngress(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "updated bastion security group to allow ssh access")
+	}
+
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "creating bastion instance")
+
+		i := &ec2.RunInstancesInput{
+			ImageId:      aws.String("ami-015e6cb33a709348e"),
+			InstanceType: aws.String("t2.micro"),
+			MaxCount:     aws.Int64(1),
+			MinCount:     aws.Int64(1),
```

Comment:
MaxCound & MinCount? Is it ASG?

Reply:
Don't know. This is the `RunInstances` API.